src/crypto/crypto_context.cc

```diff
@@ -222,6 +226,232 @@ unsigned long LoadCertsFromFile(  // NOLINT(runtime/int)
   }
 }
 
+enum TrustStatus { UNSPECIFIED, TRUSTED, DISTRUSTED };
+
+std::string getX509Name(X509_NAME* x509_name) {
+  ClearErrorOnReturn clearErrorOnReturn;
+  if (x509_name == nullptr) return {};
+
+  BIO* bio = BIO_new(BIO_s_mem());
+  if (bio == nullptr) {
+    return nullptr;
+  }
+
+  if (X509_NAME_print_ex(
+          bio, x509_name, 0, XN_FLAG_ONELINE) <= 0) {
+    return {};
+  }
+
+  const int resultLen = BIO_pending(bio);
+  char* issuer = reinterpret_cast<char *>(calloc(resultLen + 1, 1));
+  BIO_read(bio, issuer, resultLen);
+  BIO_free_all(bio);
+
+  std::string strResult = issuer;
+  return strResult;
+}
+
+bool IsSelfSigned(X509* cert) {
+  auto issuerName = getX509Name(X509_get_issuer_name(cert));
```

Comment:
I think you can simply wraps it with an `ncrypto::X509View` and use `getSubject()` etc. to get the data and then compare the content of the two.

Reply:
Any pointers? I'm struggling to find where/ how an example of how one gets an `ncrypto::X509View` (I'll keep looking though).

Reply:
I've tried:

```
ncrypto::X509View x509_view = ncrypto::X509View::From(cert);
```

But that takes an `SSLPointer&` which I haven't figured out yet

Reply:
Using `ncrypto::X509View x509_view(cert);` seems to compile although not quite sure on the next steps.

currently at:

```c++
  ncrypto::X509View x509_view(cert);
  auto subject = x509_view.getSubject();
  auto issuer = x509_view.getIssuer();

  if (strncmp(subject, issuer, 30) == 0) {
    fprintf(stderr, "Self-signed certificate detected\n");
    return true;
  } else {
    fprintf(stderr, "Self-signed certificate detected\n");
    return false;
  }
```

Reply:
Something like this? https://github.com/nodejs/node/pull/56599/commits/9cb41b051151373b2e3538bfb5f1feb6690f663b

Reply:
Not sure why I would `strncmp` here as the string could be the same up to a point but still differ after. But I could be misunderstanding.

Reply:
Actually, I realized that we can simply use `X509_issuer_name_cmp` and `X509_subject_name_cmp` and save all these?

Reply:
Thanks thats a lot simpler: https://github.com/nodejs/node/pull/56599/commits/e5254655fe512bfc0d92e6c23bbc7e577da6a47b